opensearch/Dockerfile

```diff
@@ -13,7 +13,18 @@ RUN sed -i 's#throw new OpenSearchException("GenerativeQAResponseProcessor faile
     search-processors/src/main/java/org/opensearch/searchpipelines/questionanswering/generative/GenerativeQAResponseProcessor.java
 RUN ./gradlew assemble
 
-FROM opensearchproject/opensearch:$TAG as install_ml_commons
+FROM gradle:jdk17 AS build_rps_plugin
+
+WORKDIR /build
+RUN git clone https://github.com/aryn-ai/opensearch-remote-processor.git
```

Comment:
`git clone -b 2.x --depth 1 https://github.com/aryn-ai/opensearch-remote-processor.git`

We don't care about history. Same fix above for ml-commons.git.  

Probably also remove the TAG variable at the top and just inline it. It's no longer safe with the 2.12 checkout for ml-commons. Alternately make a bunch of variables and apply them throughout, but since we don't test for other combinations it doesn't seem useful to support them.